node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1339,7 +1339,7 @@ impl BlockSignaturesCollector {
     ) -> bool {
         self.0.as_ref().map_or(false, |sigs| {
             are_signatures_sufficient_for_sync_to_genesis(
-                consensus::check_sufficient_finality_signatures(
+                check_sufficient_finality_signatures_for_sync_to_genesis(
```

Comment:
I think this is wrong: this will make us stop requesting additional signatures from other peers when we reach `ftt` instead of `(1+ftt)/2`. I think the sufficiency condition in the collector should _still_ be `(1+ftt)/2`, and we should only re-check if we have at least `ftt` of them after the loop over all the peers exits in `fetch_and_store_finality_signatures_by_block_header`.

Reply:
TBH, I'm not sure if we ever discussed it in that much detail, but I believe that's the most reasonable approach to supporting a weaker threshold, but still aiming for the stronger one in most cases.

Reply:
In e6988fea4e6d3f50ba54831217a97a71263b6d68, the algorithm has been updated to work like described:

- for block header
  - fetch signatures
    - if from storage, continue to next block (trusting there is sufficient set of signatures in the storage)
    - if from peer, check for default sufficiency
      - if sufficient, store and continue to next block
      - if not sufficient, continue with next peer
        - if out of peers, check that what we fetched so far is sufficient according to the loosened requirement
          - if yes, store
          - if `TooManySignatures` error - store as well
          - if no, bubble up the error (`InsufficientWeightForFinality` or `BogusValidator` _(the latter shouldn't happen at this stage)_)